src/components/landing-layout/navigation/sign-in-button.tsx

```diff
@@ -4,24 +4,40 @@
  * SPDX-License-Identifier: AGPL-3.0-only
  */
 
-import React from 'react'
+import equal from 'fast-deep-equal'
+import React, { useEffect, useRef, useState } from 'react'
 import { Button } from 'react-bootstrap'
 import { ButtonProps } from 'react-bootstrap/Button'
 import { Trans, useTranslation } from 'react-i18next'
 import { useSelector } from 'react-redux'
 import { LinkContainer } from 'react-router-bootstrap'
 import { ApplicationState } from '../../../redux'
 import { ShowIf } from '../../common/show-if/show-if'
+import { getApiUrl } from '../../../api/utils'
+import { INTERACTIVE_LOGIN_METHODS } from '../../../api/auth'
 
 export type SignInButtonProps = Omit<ButtonProps, 'href'>
 
 export const SignInButton: React.FC<SignInButtonProps> = ({ variant, ...props }) => {
   const { t } = useTranslation()
-  const anyAuthProviderActive = useSelector((state: ApplicationState) => Object.values(state.config.authProviders)
-                                                                               .includes(true))
+  const authProviders = useSelector((state: ApplicationState) => state.config.authProviders, equal)
+  const [loginLink, setLoginLink] = useState('/login')
+  const authEnabled = useRef(Object.values(authProviders).includes(true))
+
+  useEffect(() => {
+    const activeProviders = Object.entries(authProviders)
+                                  .filter((entry: [string, boolean]) => entry[1])
+                                  .map(entry => entry[0])
+    const activeOneClickProviders = activeProviders.filter(entry => !INTERACTIVE_LOGIN_METHODS.includes(entry))
+
+    if (activeProviders.length === 1 && activeOneClickProviders.length === 1) {
+      setLoginLink(`${ getApiUrl() }/auth/${ activeOneClickProviders[0] }`)
+    }
+  }, [authProviders, setLoginLink])
```

Comment:
Why are you using a state here and not a memo?

Reply:
Changed